### PR TITLE
acc: add to-tag for locally generated replies

### PR DESCRIPTION
**Pull request by jchavanton**
#### Pre-Submission Checklist

- [x] Commit message has the format required by CONTRIBUTING guide
- [x] Commits are split per component (core, individual modules, libs, utils, ...)
- [x] Each component has a single commit (if not, squash them into one commit)
- [x] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [x] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [ ] PR should be backported to stable branches
- [x] Tested changes locally
- [ ] Related to issue #XXXX (replace XXXX with an open issue number)

#### Description
When doing t_reply/send_reply stateful, the ACC module will not add the to-tag to the ACC events.
This patch will fix this and ensure that we get the to-tag that was locally generated used in the reply.

example :

```
modparam("acc", "acc_prepare_always", 1)
modparam("acc_json", "acc_missed_flag", 6)

failure_route[test] {
    setflag(6); // MISSED_ACC, the failure will be recorded by ACC
    t_flush_flags(); // This will set the flags in the newly created transaction
    t_reply("500", "Service Unavailable");
}
```
This does generate an ACC event, however the to-tag is not populated, my guess this is because the TM callback is done before the to-tag is generated.

This patch will populate the to-tag immediately.

### CODE REVIEW

File: src/modules/acc/acc_logic.c [hunk 1]
```diff
@@ -79,21 +79,29 @@ struct acc_enviroment acc_env;
...
 
 static inline void env_set_to(struct hdr_field *to)
 {
 	acc_env.to = to;
 }
 
+// static inline void env_set_totag(str *to_tag)
+// {
+// 	acc_env.to_tag = to_tag;
+// }
```
Comment (miconda) [inline src/modules/acc/acc_logic.c:104]: I guess this part can be removed if not intended to be used.

File: src/modules/acc/acc_logic.c [hunk 2]
```diff
@@ -372,6 +380,9 @@ void acc_onreq( struct cell* t, int type, struct tmcb_params *ps )
 	int tmcb_types;
 	int is_invite;
 
+	LM_ERR("is_mc_on: %d\n", is_mc_on(ps->req));
+	LM_ERR("is_acc_on: %d\n", is_acc_on(ps->req));
+	LM_ERR("is_acc_prepare_on: %d\n", is_acc_prepare_on(ps->req));
```
Comment (miconda) [inline src/modules/acc/acc_logic.c:385]: There are couple of error messages which I guess are only for testing purposes. Before merging should be removed. I could spot in two other places, besides the three here.

**Comment by jchavanton** [comment]
Sorry, about the delay, I left it unfinished before taking a long week-end off, I will push adjustments tomorrow

**Comment by miconda** [comment]
Still working on it, or it is finished from your point of view?

**Comment by jchavanton** [comment]
it is working, but as I integrate it in a complicated routing configuration, I may find more little modifications.

Just found one ...

We could wait a few more days.

**Comment by jchavanton** [comment]
Would be good if I could also improve this to always have the real reason phrase sent.
Having a look at the options.

```
acc_env.code = code;
if (reply==FAKED_REPLY || reply==NULL) {
        /* code */
        acc_env.code_s.s =
                int2bstr((unsigned long)code, code_buf, &acc_env.code_s.len);
        /* reason */
        acc_env.reason.s = error_text(code);                                  
        acc_env.reason.len = strlen(acc_env.reason.s);
```

**Comment by jchavanton** [comment]
not expecting more modifications based on testing.
ready to review/merge

**Comment by miconda** [comment]
Thanks!

**Pull request merged**